### PR TITLE
install-deps: Force remove ceph-libboost* packages

### PR DESCRIPTION
**Pull request by rosinL**
As `apt-get remove ceph-libboost*` show the error and cause the make check failed. `E: The package ceph-libboost1.79-dev needs to be reinstalled, but I can't find an archive for it.` The ceph-libboost* packages may be not removed, so use `dpkg` to force remove ceph-libboost*.

Fixes: https://tracker.ceph.com/issues/64032

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by rosinL** [comment]
@idryomov @adamemerson @cbodley ,please take a review, thanks

File: install-deps.sh [hunk 1]
```diff
@@ -177,6 +177,14 @@ function clean_boost_on_ubuntu {
     # so no need to spare it.
     if test -n "$installed_ver"; then
 	$SUDO env DEBIAN_FRONTEND=noninteractive apt-get -y --fix-missing remove "ceph-libboost*"
+	# When an error occurs during `apt-get remove ceph-libboost*`, ceph-libboost* packages
+	# may be not removed, so use `dpkg` to force remove ceph-libboost*.
+	local ceph_libboost_pkgs=$(dpkg -l | grep ceph-libboost* | awk '{print $2}' |
+		                        awk -F: '{print $1}')
+	if test -n "$ceph_libboost_pkgs"; then
+	    ci_debug "Force remove ceph-libboost* packages $ceph_libboost_pkgs"
+	    $SUDO dpkg --purge --force-all $ceph_libboost_pkgs
```
Comment (idryomov) [inline install-deps.sh:186]: I'm not into packaging at all, but I'm wondering if both `apt-get remove` and `dpkg --purge` are needed.  If `dpkg --purge` is stronger, perhaps we should just switch to it?

**Review by idryomov — COMMENTED** [review]
Did you verify that the error hit in https://tracker.ceph.com/issues/64032 is addressed by this change?  If so, how?

File: install-deps.sh (hunk 1 above)
Comment (rosinL) [reply install-deps.sh:186]: `apt-get remove` can handle dependencies and remove the related softwares,  It is more safe. In this case, we can just use `dpkg --purge`

**Comment by rosinL** [comment]
> Did you verify that the error hit in https://tracker.ceph.com/issues/64032 is addressed by this change? If so, how?

Yes, the error packages can be force removed.

**Review by ljflores — APPROVED** [review]

**Comment by cbodley** [comment]
interesting that `make check (arm64)` succeeded despite unit test failures in [the log](https://jenkins.ceph.com/job/ceph-pull-requests-arm64/50792/consoleFull):
> The following tests FAILED:
	 57 - unittest_mempool (Failed)
	250 - unittest-transaction-manager (Timeout)
	256 - unittest-omap-manager (Timeout)

**Comment by ljflores** [comment]
> interesting that `make check (arm64)` succeeded despite unit test failures in [the log](https://jenkins.ceph.com/job/ceph-pull-requests-arm64/50792/consoleFull):
> 
> > The following tests FAILED:
> > 57 - unittest_mempool (Failed)
> > 250 - unittest-transaction-manager (Timeout)
> > 256 - unittest-omap-manager (Timeout)

Yes, someone noted this awhile back. Would be good to fix this so the check is more meaningful. I added a ticket for that here: https://tracker.ceph.com/issues/64117

**Pull request merged**